src/flatfiledirmgr.rs

```diff
@@ -70,13 +80,37 @@ impl<R: Runtime> FlatFileDirMgr<R> {
         }))
     }
 
+    /// Check cache directory content.
+    fn check_directory(cache_path: &Path) -> Result<()> {
+        let missing_files: Vec<&&str> = [
+            CONSENSUS_FILENAME,
+            MICRODESCRIPTORS_FILENAME,
+            CERTIFICATE_FILENAME,
+            CHURN_FILENAME,
+        ]
+        .iter()
+        .filter(|filename| !cache_path.join(filename).exists())
+        .collect();
+        if !missing_files.is_empty() {
+            let pprinted_missing_files = missing_files
+                .iter()
+                .map(|filename| filename.to_string())
+                .collect::<Vec<_>>()
+                .join(" ");
+            debug!("required file(s) missing: {}", pprinted_missing_files);
```

Comment:
In fact the formatter has some special characters, see https://doc.rust-lang.org/std/fmt/. Most of the structures implement the `Debug` trait, which can be called using `{variable:?}`:

```suggestion
            debug!("required file(s) missing: {missing_files:?}");
```

Reply:
but won't that look like this, e.g. `required file(s) missing: [certificate.txt, authority.json]`? instead of pretty `required file(s) missing: certificate.txt authority.json`

Reply:
updated it